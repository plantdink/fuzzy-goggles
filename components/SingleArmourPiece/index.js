export * from './SingleArmourPiece';
export { default } from './SingleArmourPiece';
