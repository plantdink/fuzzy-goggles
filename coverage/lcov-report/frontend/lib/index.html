
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for frontend/lib</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> frontend/lib</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.61% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>55/65</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">64% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>16/25</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">71.42% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>20/28</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.78% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>39/46</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="abbreviationsDictionary.js"><a href="abbreviationsDictionary.js.html">abbreviationsDictionary.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="displayNumbers.js"><a href="displayNumbers.js.html">displayNumbers.js</a></td>
	<td data-value="90.9" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 90%"></div><div class="cover-empty" style="width: 10%"></div></div>
	</td>
	<td data-value="90.9" class="pct high">90.9%</td>
	<td data-value="11" class="abs high">10/11</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	</tr>

<tr>
	<td class="file high" data-value="displayStrings.js"><a href="displayStrings.js.html">displayStrings.js</a></td>
	<td data-value="84.61" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 84%"></div><div class="cover-empty" style="width: 16%"></div></div>
	</td>
	<td data-value="84.61" class="pct high">84.61%</td>
	<td data-value="13" class="abs high">11/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	</tr>

<tr>
	<td class="file medium" data-value="testUtils.js"><a href="testUtils.js.html">testUtils.js</a></td>
	<td data-value="74.07" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 74%"></div><div class="cover-empty" style="width: 26%"></div></div>
	</td>
	<td data-value="74.07" class="pct medium">74.07%</td>
	<td data-value="27" class="abs medium">20/27</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="56.25" class="pct medium">56.25%</td>
	<td data-value="16" class="abs medium">9/16</td>
	<td data-value="68.75" class="pct medium">68.75%</td>
	<td data-value="16" class="abs medium">11/16</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Apr 05 2022 07:32:49 GMT+1000 (Australian Eastern Standard Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    