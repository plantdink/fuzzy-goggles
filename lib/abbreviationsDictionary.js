export function deAbbreviate(abbreviation) {
  let fullTerm = '';
  if (abbreviation === 'OOC') fullTerm = 'Damage to Targets out of Cover';
  if (abbreviation === 'DTH') fullTerm = 'Damage to Health';
  if (abbreviation === 'DTA') fullTerm = 'Damage to Armour';
  if (abbreviation === 'CHC') fullTerm = 'Critical Hit Chance';
  if (abbreviation === 'CHD') fullTerm = 'Critical Hit Damage';
  if (abbreviation === 'HSD') fullTerm = 'Head Shot Damage';
  return fullTerm;
}
