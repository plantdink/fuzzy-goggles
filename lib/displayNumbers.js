export function displayPercentage(number = 0) {
  const displayNumber = number / 10;
  return displayNumber;
}

export function hsDisplay(number = 0) {
  const displayNumber = number / 100;
  return displayNumber;
}

export function millisecondsToSeconds(number = 0) {
  const displayNumber = ((number % 60000) / 1000).toFixed(2);
  return displayNumber;
}

export function humanReadableNumber(num = 0, lang = null) {
  if (!num) return;
  const locale = lang || 'en'; // removed || document.documentElement.lang
  const number = parseFloat(num, 10);
  return number.toLocaleString(locale);
}
