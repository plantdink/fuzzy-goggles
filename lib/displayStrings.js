export function lowerCaseFirstLetter(string) {
  const lowerCaseString = string.toLowerCase();
  const titleString = lowerCaseString.replace(
    /(^\w{1})|(\s+\w{1})/g,
    (letter) => letter.toUpperCase()
  );
  const noWhiteSpaceString = titleString.replaceAll(/\s/g, '');
  const finalLink =
    noWhiteSpaceString.charAt(0).toLowerCase() + noWhiteSpaceString.slice(1);
  return finalLink;
}

export function titleCase(string) {
  return string
    .toLowerCase()
    .split(' ')
    .map(function (word) {
      return word.charAt(0).toUpperCase() + word.slice(1);
    })
    .join(' ');
}

export function stringToParagraphs(string) {
  const sentences = string.split(/(?<=\.)/g);
  return sentences.map((sentence, index) => <p key={index}>{sentence}</p>);
}

export function lowerCaseLink(string) {
  const stringToEdit = string[0].toLowerCase() + string.slice(1) || string;
  return stringToEdit;
}
