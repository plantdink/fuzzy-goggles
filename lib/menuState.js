import { createContext, useContext, useState } from 'react';

const LocalStateContext = createContext();
const LocalStateProvider = LocalStateContext.Provider;

function MenuStateProvider({ children }) {
  const [menuOpen, setMenuOpen] = useState(false);

  function toggleMenu() {
    setMenuOpen(!menuOpen);
  }

  function closeMenu() {
    setMenuOpen(false);
  }

  function openMenu() {
    setMenuOpen(true);
  }

  return (
    <LocalStateProvider
      value={{
        menuOpen,
        setMenuOpen,
        toggleMenu,
        closeMenu,
        openMenu,
      }}
    >
      {children}
    </LocalStateProvider>
  );
}

// custom hook for accessing menu local state
function useMenu() {
  const all = useContext(LocalStateContext);
  return all;
}
export { MenuStateProvider, useMenu };
