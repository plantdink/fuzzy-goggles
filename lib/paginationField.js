import { WEAPON_PAGINATION_QUERY } from '../queries/PaginationQueries';

export default function paginationField() {
  // apollo will ask read for items.
  // It will return them from the cache if they exist or
  // it will return false & trigger a network request
  return {
    keyArgs: false, // toggle for apollo to let this function handle things
    read(existing = [], { args, cache }) {
      const { skip, first } = args;

      // read the number of items on the page from the cache
      const data = cache.readQuery({ query: WEAPON_PAGINATION_QUERY });
      const count = data?._allWeaponsMeta?.count;
      const page = skip / first + 1;
      const pages = Math.ceil(count / first);

      // check for existing items
      const items = existing.slice(skip, skip + first).filter((x) => x);

      // if there are items & there are not enough for the requested amount & it is the last page, go to the network
      if (items.length && items.length !== first && page === pages) {
        return items;
      }
      // there are no items, request from network
      if (items.length !== first) {
        return false;
      }

      // if there are items in cache, return them
      if (items.length) {
        return items;
      }

      // fallback to network
      return false;
    },

    // apollo will run this when the network request returns with items
    merge(existing, incoming, { args }) {
      const { skip, first } = args;
      const merged = existing ? existing.slice(0) : [];
      for (let i = skip; i < skip + incoming.length; ++i) {
        merged[i] = incoming[i - skip];
      }

      // return the merged items from the cache
      return merged;
    },
  };
}
