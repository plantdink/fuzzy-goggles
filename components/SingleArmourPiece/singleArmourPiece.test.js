import { render, screen, cleanup } from '@testing-library/react';
import SingleArmourPiece from '.';
import { fakeArmourPiece } from '../../lib/testUtils';

const armourPiece = fakeArmourPiece();

afterEach(cleanup);

describe('<SingleArmourPiece />', () => {
  it('renders correctly for a single armour piece', async () => {
    const { container, debug } = render(
      <SingleArmourPiece armourPiece={armourPiece} />
    );

    await screen.findByTestId('singleArmourPiece');
    expect(container).toMatchSnapshot();

    // Move to SingleBrandPage test
    // const testArmourPieceTitle = screen.getByText(armourPiece.name);
    // expect(testArmourPieceTitle).toBeInTheDocument();

    // check World Tier 5 stats render if present
    //     if (fakeQuery.is)
    // const worldTier5Check = screen.getByRole("columnheader", {
    //   name: /world tier 5 stats/i,
    // });

    const testArmourPieceIcon = await screen.findByTestId('itemIcon');
    expect(testArmourPieceIcon).toBeInTheDocument();

    const testPiece = await screen.findByTestId('itemPiece');
    if (armourPiece.piece !== '') expect(testPiece).toBeInTheDocument();
    expect(testPiece).toHaveTextContent(armourPiece.piece);

    const testNotes = await screen.findByTestId('itemNotes');
    if (armourPiece.notes !== '') expect(testNotes).toBeInTheDocument();
    expect(testNotes).toHaveTextContent(armourPiece.notes);
  });
});
