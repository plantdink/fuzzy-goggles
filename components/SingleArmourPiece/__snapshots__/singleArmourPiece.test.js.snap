// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleArmourPiece /> renders correctly for a single armour piece 1`] = `
<div>
  <img
    alt="Mask icon"
    class="single-gear-item__icon-image"
    data-testid="itemIcon"
    src="https://res.cloudinary.com/dbm1h5w02/image/upload/v1647498165/agent-field-manual/6232d3b4ec80b2199ff2c03a.png"
  />
  <div>
    <p
      data-testid="itemPiece"
    >
      Mask
    </p>
    <p
      data-testid="itemNotes"
    >
      This mask has been manufactured by Gila Guard and comes with Armour as the primary Core attribute. It has one fixed minor attribute - 100% Scanner Pulse Haste - and one random minor attribute which can be any of the stats listed below. It also has one gear mod slot.
    </p>
  </div>
  <div
    class="single-gear-item__content"
    data-testid="singleArmourPiece"
  >
    <div
      class="single-gear-item__details"
    >
      <table
        data-testid="singleYellowArmourPiece"
      >
        <thead>
          <tr
            class="single-gear-item__title-bar"
          >
            <th
              class="single-gear-item__subheading"
            >
              Level 40 Stats
            </th>
          </tr>
          <tr
            class="single-gear-item__title-bar"
          >
            <th
              class="single-gear-item__subheading"
            >
              Core Attributes
            </th>
          </tr>
          <tr>
            <td />
            <td>
              Max Value
            </td>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="itemCoreTwoType"
          >
            <td>
              Armour
            </td>
            <td>
              +
               
              170,000
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead
          class="single-gear-item__title-bar"
        >
          <tr>
            <th
              class="single-gear-item__h3subheading"
            >
              Offensive Minor Attributes
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="itemOffAttrOneMax40Value"
          >
            <td>
              Critical Hit Chance
            </td>
            <td>
              +
               
              6
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemOffAttrTwoMax40Value"
          >
            <td>
              Critical Hit Damage
            </td>
            <td>
              +
               
              15
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemOffAttrThreeMax40Value"
          >
            <td>
              Headshot Damage
            </td>
            <td>
              +
               
              10
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemOffAttrFourMax40Value"
          >
            <td>
              Weapon Handling
            </td>
            <td>
              +
               
              8
               
              %
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead
          class="single-gear-item__title-bar"
        >
          <tr>
            <th
              class="single-gear-item__h3subheading"
            >
              Defensive Minor Attributes
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="itemDefAttrOneMax40Value"
          >
            <td>
              Armour Regeneration
            </td>
            <td>
              +
               
              4,925
               
              per second
            </td>
          </tr>
          <tr
            data-testid="itemDefAttrTwoMax40Value"
          >
            <td>
              Explosive Resistance
            </td>
            <td>
              +
               
              10
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemDefAttrThreeMax40Value"
          >
            <td>
              Hazard Protection
            </td>
            <td>
              +
               
              10
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemDefAttrFourMax40Value"
          >
            <td>
              Health
            </td>
            <td>
              +
               
              18,935
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead
          class="single-gear-item__title-bar"
        >
          <tr>
            <td
              class="single-gear-item__h3subheading"
            >
              Utility Minor Attributes
            </td>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="itemUtilAttrOneMax40Value"
          >
            <td>
              Skill Damage
            </td>
            <td>
              +
               
              10
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemUtilAttrTwoMax40Value"
          >
            <td>
              Skill Haste
            </td>
            <td>
              +
               
              12
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemUtilAttrThreeMax40Value"
          >
            <td>
              Skill Repair
            </td>
            <td>
              +
               
              20
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemUtilAttrFourMax40Value"
          >
            <td>
              Status Effects
            </td>
            <td>
              +
               
              10
               
              %
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="single-gear-item__details"
    >
      <table
        data-testid="singleYellowArmourPieceWT5"
      >
        <thead>
          <tr
            class="single-gear-item__title-bar"
          >
            <th
              class="single-gear-item__subheading"
            >
              World Tier 5 Stats
            </th>
          </tr>
          <tr
            class="single-gear-item__title-bar"
          >
            <th
              class="single-gear-item__subheading"
            >
              Core Attributes
            </th>
          </tr>
          <tr>
            <td />
            <td>
              Max Value
            </td>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="itemCoreTwoTypeWT5"
          >
            <td>
              Armour
            </td>
            <td>
              +
               
              42,458
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead
          class="single-gear-item__title-bar"
        >
          <tr>
            <th
              class="single-gear-item__h3subheading"
            >
              Offensive Minor Attributes
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="itemOffAttrOneMaxWT5Value"
          >
            <td>
              Critical Hit Chance
            </td>
            <td>
              +
               
              5.1
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemOffAttrTwoMaxWT5Value"
          >
            <td>
              Critical Hit Damage
            </td>
            <td>
              +
               
              7.9
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemOffAttrThreeMaxWT5Value"
          >
            <td>
              Headshot Damage
            </td>
            <td>
              +
               
              6.9
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemOffAttrFourMaxWT5Value"
          >
            <td>
              Weapon Handling
            </td>
            <td>
              +
               
              7
               
              %
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead
          class="single-gear-item__title-bar"
        >
          <tr>
            <th
              class="single-gear-item__h3subheading"
            >
              Defensive Minor Attributes
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="itemDefAttrOneMaxWT5Value"
          >
            <td>
              Armour Regeneration
            </td>
            <td>
              +
               
              1,223
               
              per second
            </td>
          </tr>
          <tr
            data-testid="itemDefAttrTwoMaxWT5Value"
          >
            <td>
              Explosive Resistance
            </td>
            <td>
              +
               
              6.6
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemDefAttrThreeMaxWT5Value"
          >
            <td>
              Hazard Protection
            </td>
            <td>
              +
               
              6.6
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemDefAttrFourMaxWT5Value"
          >
            <td>
              Health
            </td>
            <td>
              +
               
              4,695
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead
          class="single-gear-item__title-bar"
        >
          <tr>
            <td
              class="single-gear-item__h3subheading"
            >
              Utility Minor Attributes
            </td>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="itemUtilAttrOneMaxWT5Value"
          >
            <td>
              Skill Damage
            </td>
            <td>
              +
               
              6.9
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemUtilAttrTwoMaxWT5Value"
          >
            <td>
              Skill Haste
            </td>
            <td>
              +
               
              7.9
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemUtilAttrThreeMaxWT5Value"
          >
            <td>
              Skill Repair
            </td>
            <td>
              +
               
              12.1
               
              %
            </td>
          </tr>
          <tr
            data-testid="itemUtilAttrFourMaxWT5Value"
          >
            <td>
              Status Effects
            </td>
            <td>
              +
               
              6.6
               
              %
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    class="single-gear-item__title-bar"
  >
    <h3
      class="single-gear-item__h3subheading"
    >
      Number of Mod Slots
    </h3>
  </div>
  <ul>
    <li>
      1
    </li>
  </ul>
</div>
`;
